addons/mail/static/src/components/composer/composer.xml

```diff
@@ -20,20 +20,36 @@
                         t-ref="dropzone"
                     />
                 </t>
-                <FileUploader
-                    attachmentLocalIds="composer.attachments.map(attachment => attachment.localId)"
-                    isPending="true"
-                    newAttachmentExtraData="newAttachmentExtraData"
-                    threadLocalId="composer.thread.localId"
-                    t-ref="fileUploader"
-                />
+                <t t-if="composer.thread">
+                    <FileUploader
+                        attachmentLocalIds="composer.attachments.map(attachment => attachment.localId)"
+                        isPending="true"
+                        newAttachmentExtraData="newAttachmentExtraData"
+                        threadLocalId="composer.thread.localId"
+                        t-ref="fileUploader"
+                    />
+                </t>
+                <t t-if="composer.messageInEditing">
+                    <FileUploader
+                        attachmentLocalIds="composer.attachments.map(attachment => attachment.localId)"
+                        isPending="true"
+                        newAttachmentExtraData="newAttachmentExtraData"
+                        threadLocalId="composer.messageInEditing.originThread.localId"
+                        t-ref="fileUploader"
+                    />
+                </t>
                 <t t-if="hasHeader">
                     <div class="o_Composer_coreHeader text-truncate">
                         <t t-if="props.hasThreadName and composer.thread">
                             <span class="o_Composer_threadName">
                                 on: <b><t t-esc="composer.thread.displayName"/></b>
                             </span>
                         </t>
+                        <t t-if="props.hasThreadName and composer.messageInEditing">
+                            <span class="o_Composer_threadName">
+                                on: <b><t t-esc="composer.messageInEditing.originThread.displayName"/></b>
```

Comment:
I don't think it's safe to assume we are aware of the `originThread` of the `message`, even if all messages have an `originThread` currently.

I think it's safe thanks to how the feature of message edition requires to display the message in the context of thread, and the fetched data have the `originThread` info... But it's not safe for the future if we don't guard origin thread knowledge here.